tc/src/main/java/com/epam/tc/dao/category/CategoryDaoImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.tc.dao.category;
+
+import com.epam.tc.dao.CRUDdaoImpl;
+import com.epam.tc.model.Category;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.PersistenceContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class CategoryDaoImpl extends CRUDdaoImpl<Category> implements CategoryDao {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CategoryDaoImpl.class);
+
+    @PersistenceContext
```

Comment:
It's shouldn't be @Autowired, because got from library of Hibernate.javax.persistence and importing in class